catalyst/dl/callbacks/base.py

```diff
@@ -102,6 +102,9 @@ def on_epoch_end(self, state: RunnerState):
         if state.stage.startswith("infer"):
             return
 
+        extra_checkpoint_data = state.extra_checkpoint_data \
+            if hasattr(state, 'extra_checkpoint_data') else {}
```

Comment:
let's just add `checkpoint_data` to state init, so we can use it also from Config API.
proposed way:
```
state_params:
    checkpoint_data:
        exp: "thin is my best experiments"
        date: "today"
```

This will also reduce such `hasattr` check.